### PR TITLE
Enable initialisation and estimation of optional spatio-temperal coupling parameters to gaussian profile

### PR DESCRIPTION
**Pull request by huixingjian**
### PR Description

This PR, synergized with PR #182, introduces the initialization and estimation of first-order spatio-temporal coupling parameters in the ```GaussianLongitudinalProfile``` class. Four additional parameters have been added to the constructor:

- Group-delay dispersion `phi2` in $s^2$ parameterized by: $\varphi^{(2)} = \frac{dt_0}{d\omega}$, 
- Spatial chirp `zeta` in $m \\cdot s$ parameterized by: $\zeta = \frac{dx_0}{d\omega}$(Here $x_0$ is the beam center position), 
- Angular dispersion `beta` in $s$  parameterized by: $\beta = \frac{d\theta_0}{d\omega}$( Here $\theta_0$ is the propagation angle of this component.)
- Finally `stc_theta` is the direction of transverse spatio-coupling in xoy plane

### Issue addressed

With the introduction of spatio-temporal coupling (STC), the ```LongitudinalProfile``` class can leverage additional information from the transverse profile. Along with the aforementioned parameters, the waist and focus length are also incorporated into the constructor. The transverse points x and y have been set to default values of 0 in the evaluate member function.

In the laser utilities, a new function, ```get_STC```, has been added to evaluate the STC parameters using the following formulas, with the laser envelope expressed as $a = a_0 e^{i\theta}$:
- The temporal chirp is calculated through
  $\\phi^{(2)} = \\frac{4\\varphi^{(2)}}{4(\\varphi^{(2)})^2+\\tau^4}$
  Here $\\tau$ is duration in **_s_**, and `phi2` refers to group-delay dispersion $\\phi^{(2)} $, and $\\Phi^{(2)}$ can be calculated by $\\frac{\\partial^2 \\theta }{\\partial t ^2}$.
- Similarly, the spatial chirp is tested through:
  $\\nu = \\frac{4\\zeta }{w_0^2\\tau^2+4\\zeta^2}$
  Here $L_0$ and $w_0$ are the laser duration and laser waist respectively. $\zeta = \frac{dr_0}{d\omega}$, where $r_0$  is the transverse beam centroid ($\\theta$ is the ```stc_theta```mentioned above). 
- Finally, the angular chirp term is tested through:
  $\\beta = \\frac{d_{\\theta_0}}{ d\\omega}$
  with $\\theta_0$ refering to the propagating angle of the frequency component

### Benefits

First order STC is enabled on both the initialisation and evaluation sides.

### Test
The initialisation and the utils are tested through CI test ```test_STC.py```

### CODE REVIEW

File: lasy/profiles/gaussian_profile.py [hunk 1]
```diff
@@ -61,6 +61,36 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
     z_foc : float (in meter), optional
         Position of the focal plane. (The laser pulse is initialized at `z=0`.)
 
+    beta : float (in second), optional
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
    beta : float (in second), optional (default '0')
```

File: lasy/profiles/gaussian_profile.py [hunk 2]
```diff
@@ -61,6 +62,36 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
...
+
+    phi2 : float (in second^2), optional (default '0')
+        The group-delay dispersion parameterized by:
+
+        .. math::
+
+            \phi^{(2)} = \frac{dt}{d\omega}
+
+    zeta : float (in meter * second) optional (default '0')
+        The spatio-chirp parameterized by:
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
        The spatial chirp, parameterized by:
```

File: lasy/profiles/gaussian_profile.py [hunk 3]
```diff
@@ -61,6 +62,36 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
...
+
+        .. math::
+
+             \zeta = \frac{dx_0}{d\omega}
+
+        Here :math:`x_0` is the beam center position.
+
+    stc_theta :  float (in rad) optional (default '0')
+        Transverse direction along which spatio-temporal field couples.
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
        Transverse direction along which there are chirps and spatio-temporal couplings.
```

File: lasy/profiles/longitudinal/gaussian_profile.py
Comment (MaxThevenet) [inline lasy/profiles/longitudinal/gaussian_profile.py:1]: This profile should not be changed, or just marginally. This implementation should be in `profiles/gaussian_profile.py`.

File: lasy/profiles/longitudinal/longitudinal_profile.py
Comment (MaxThevenet) [inline lasy/profiles/longitudinal/longitudinal_profile.py:1]: This profile should not be changed.

File: tests/test_STC.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    wavelength=0.6e-6,  # m
+    pol=(1, 0),
+    laser_energy=1.0,  # J
+    w0=5e-6,  # m
+    tau=5e-14,  # s
+    t_peak=0.0,  # s
+    beta=3e-18,
+    zeta=2.4e-24,
+    phi2=2.4e-19,
+    stc_theta=scc.pi / 2,
```
Comment (MaxThevenet) [inline tests/test_STC.py]: Could you make the style consistent?

File: tests/test_STC.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    lo=(-10e-6, -10e-6, -10e-14),
+    hi=(10e-6, 10e-6, +10e-14),
+    npoints=(50, 60, 70),
+    profile=profile,
+)
+# Create laser with given profile in `rt` geometry.
+laser_2d = Laser(
+    dim="rt",
+    lo=(-10e-6, -10e-14),
+    hi=(10e-6, +10e-14),
```
Comment (MaxThevenet) [inline tests/test_STC.py]: style consistency

File: tests/test_STC.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    npoints=(60, 70),
+    profile=profile,
+)
+STC_3d = get_STC(laser_3d.dim, laser_3d.grid, k0=2 * scc.pi / 0.6e-6)
+STC_2d = get_STC(laser_2d.dim, laser_2d.grid, k0=2 * scc.pi / 0.6e-6)
+np.testing.assert_approx_equal(STC_2d["phi2"], 2.4e-19, significant=2)
+np.testing.assert_approx_equal(STC_3d["phi2"], 2.4e-19, significant=2)
+np.testing.assert_approx_equal(STC_3d["beta_y"], 3e-18, significant=2)
+np.testing.assert_approx_equal(STC_3d["zeta_y"], 2.4e-24, significant=2)
```
Comment (MaxThevenet) [inline tests/test_STC.py]: Any reason to use this over `np.isclose`? We us that one in some place.

File: lasy/utils/laser_utils.py [hunk 7]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: I would recommend to write 1 sentence per line. The pre-commit CI will fix it.

File: lasy/utils/laser_utils.py [hunk 8]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+            beta_x, beta_y: Angular dispersion in :math:` \beta = d\theta_0/d\omega`(Important note:
+                  for now beta is only correct when zeta and phi2 are 0!)
+            pft_x, pft_y: Pulse front tilt in :math:` p=dt/dx`
+    All those above units and definitions are taken from
+    `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.
+    """
+    tau = 2 * get_duration(grid, dim)
+    w0 = get_w0(grid, dim)
+
+    # Initialise the returned dictionary
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: US English was used above, and is the norm in LASY.
```suggestion
    # Initialize the returned dictionary
```

File: lasy/utils/laser_utils.py [hunk 9]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        "zeta_y": 0,
+        "beta_x": 0,
+        "beta_y": 0,
+        "pft_x": 0,
+        "pft_y": 0,
+    }
+
+    # Get temporal and spectral field
+    env = grid.get_temporal_field()
+    env_abs = np.abs(env**2)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: Could you make the naming more accurate?

File: lasy/utils/laser_utils.py [hunk 10]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        Wavenumber of the field
+
+    Return
+    ----------
+    STC_fac : dict of floats
+        A dictionary of floats corresponding to the STC factors. The keys are:
+            Phi2: Group-delayed dispersion in :math:`\Phi^{(2)}=d(\omega_0)/dt`
+            phi2: Group-delayed dispersion in :math:`\phi^{(2)}=dt_0/d(\omega)`
+            nu_x, nu_y: Spatio-chirp in :math:`\nu=d(\omega_0)/dx`
+            zeta_x, zeta_y: Spatio-chirp in :math:`\zeta=dx_0/d(\omega_0)`
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: Group Delay Dispersion (GDD) and spatial chirp.

File: lasy/utils/laser_utils.py [hunk 11]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    # Get temporal and spectral field
+    env = grid.get_temporal_field()
+    env_abs = np.abs(env**2)
+    env_spec = grid.get_spectral_field()
+    env_spec_abs = np.abs(env_spec**2)
+
+    # Get the spectral axis
+    dt = grid.dx[-1]
+    Nt = grid.shape[-1]
+    omega = 2 * np.pi * np.fft.fftfreq(Nt, dt) + k0 * c
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py:1025]: Ok, indeed this axis doesn't seem to be available from Grid. I think this should be fixed, good point!

File: lasy/utils/laser_utils.py [hunk 12]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    # Calculate group-delayed dispersion
+    phi_envelop = np.unwrap(np.array(np.arctan2(env.imag, env.real)), axis=2)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: 1D unwrapping can be a bit dodgy. Also, `np.angle` would be nicer here.

File: lasy/utils/laser_utils.py [hunk 13]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    pphi_pt = np.gradient(phi_envelop, grid.dx[-1], axis=2)
+    pphi_pt2 = np.gradient(pphi_pt, grid.dx[-1], axis=2)
+    STC_fac["Phi2"] = np.average(pphi_pt2, weights=env_abs)
+    STC_fac["phi2"] = np.max(
+        np.roots([4 * STC_fac["Phi2"], -4, tau**4 * STC_fac["Phi2"]])
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: always the right value?

File: lasy/utils/laser_utils.py [hunk 14]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        else:
+            A2 = np.r_[A2[::-1][:-1], A2]
+            ax = np.r_[-ax[::-1][:-1], ax]
+
+    sigma = 2 * np.sqrt(np.average(ax**2, weights=A2))
+
+    return sigma
+
+
+def get_STC(dim, grid, k0):
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: I think we should cut this into multiple functions, with clear names, and put the commonalities in helper functions.

File: lasy/utils/laser_utils.py [hunk 15]
```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    omega = 2 * np.pi * np.fft.fftfreq(Nt, dt) + k0 * c
+    # Calculate group-delayed dispersion
+    phi_envelop = np.unwrap(np.array(np.arctan2(env.imag, env.real)), axis=2)
+    pphi_pt = np.gradient(phi_envelop, grid.dx[-1], axis=2)
+    pphi_pt2 = np.gradient(pphi_pt, grid.dx[-1], axis=2)
+    STC_fac["Phi2"] = np.average(pphi_pt2, weights=env_abs)
+    STC_fac["phi2"] = np.max(
+        np.roots([4 * STC_fac["Phi2"], -4, tau**4 * STC_fac["Phi2"]])
+    )
+    # No spatial chirp and angular chirp in 'rt' coordinate
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: Overall everything below would benefit from more comments, to gain clarity.

**Review by MaxThevenet — CHANGES_REQUESTED** [review]
Thanks for this PR! Very nice that you can have all sorts of chirps together and can disentangle them in post-processing. See comments below, mostly about cleaning and clarifying.

**Comment by MaxThevenet** [comment]
Please check in particular https://github.com/LASY-org/lasy/pull/182/files#r1406449309

File: lasy/profiles/longitudinal/longitudinal_profile.py [hunk 16]
```diff
@@ -13,6 +13,7 @@ class LongitudinalProfile(object):
     def __init__(self, wavelength):
         self.lambda0 = wavelength
         self.omega0 = 2 * pi * c / self.lambda0
+        self.k0 = 2.0 * pi / wavelength
```
Comment (MaxThevenet) [inline lasy/profiles/longitudinal/longitudinal_profile.py]: I don't think it's used anywhere.
```suggestion
```

File: lasy/utils/laser_utils.py [hunk 17]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+     Return
+    ----------
+    phi2: Group-delayed dispersion in :math:`\Phi^{(2)}=d(\omega_0)/dt` (second^-2)
+    varphi2: Group-delayed dispersion in :math:`\varphi^{(2)}=dt_0/d(\omega)` (second^2)
+    """
+    tau = 2 * get_duration(grid, dim)
+    env = grid.get_temporal_field()
+    env_abs2 = np.abs(env**2)
+    # Calculate group-delayed dispersion
+    phi_envelop = np.unwrap(np.array(np.arctan2(env.imag, env.real)), axis=2)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: `np.angle`?
Note that this 1D unwrapping is very easily very wrong. But if that's a problem we could implement a 3D unwrapping later one, there was an example of this actually in LASY at some point.

File: lasy/utils/laser_utils.py [hunk 18]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    pphi_pt = np.gradient(phi_envelop, grid.dx[-1], axis=2)
+    pphi_pt2 = np.gradient(pphi_pt, grid.dx[-1], axis=2)
+    phi2 = np.average(pphi_pt2, weights=env_abs2)
+    varphi2 = np.max(np.roots([4 * phi2, -4, tau**4 * phi2]))
+    return phi2, varphi2
+
+
+def get_zeta(dim, grid, k0):
+    r"""
+    Calculate the spatio-chirp of the laser.
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    Calculate the spatial chirp of the laser.
```

File: lasy/utils/laser_utils.py [hunk 19]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    zeta_x, zeta_y: Spatio-chirp in :math:`\zeta=dx_0/d(\omega_0)` (meter * second)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    zeta_x, zeta_y: Spatial chirp in :math:`\zeta=dx_0/d(\omega_0)` (meter * second)
```

File: lasy/utils/laser_utils.py [hunk 20]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    nu_x, nu_y: Spatio-chirp in :math:`\nu=d(\omega_0)/dx` (meter^-1 * second^-1)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: Also, please fix space before colon for parameters everywhere!
```suggestion
    nu_x, nu_y : Spatial chirp in :math:`\nu=d(\omega_0)/dx` (meter^-1 * second^-1)
```

File: lasy/utils/laser_utils.py [hunk 21]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    """
+    assert dim == "xyt", "No spatial chirp for axis-sysmetric dimension"
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: !

File: lasy/utils/laser_utils.py [hunk 22]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    beta_x, beta_y: Angular dispersion in :math:` \beta = d\theta_0/d\omega` (second)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    beta_x, beta_y : Angular dispersion in :math:` \beta = d\theta_0/d\omega` (second)
```
and everywhere

File: lasy/utils/laser_utils.py [hunk 23]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    derivative_x_beta = np.gradient(angle_y, omega, axis=2)
+    derivative_y_beta = np.gradient(angle_x, omega, axis=2)
+    beta_x = np.average(derivative_x_beta, weights=env_spec_abs2)
+    beta_y = np.average(derivative_y_beta, weights=env_spec_abs2)
+    return [beta_x, beta_y]
+
+
+def get_pft(dim, grid):
+    r"""
+    Calculate the pulse-front-tilt dispersion of the laser.
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    Calculate the pulse front tilt (PFT) of the laser.
```

File: lasy/utils/laser_utils.py [hunk 24]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    pft_x, pft_y: Pulse front tilt in :math:` p=dt/dx` (second * meter^-1)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    pft_x, pft_y : Pulse front tilt in :math:` p=dt/dx` (second * meter^-1)
```

File: lasy/utils/laser_utils.py [hunk 25]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    """
+    assert dim == "xyt", "No pulse front tilt for axis-sysmetric dimension"
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]: Please fix all

File: lasy/utils/laser_utils.py [hunk 26]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    angle_x, angle_y: propagating angle in :math:` p=k_x or k_y/kz` (rad)
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    angle_x, angle_y : propagating angle in :math:` p=k_x or k_y/kz` (rad)
```
please fix all

**Review by MaxThevenet — CHANGES_REQUESTED** [review]
Thanks! Please see comments

File: lasy/profiles/gaussian_profile.py [hunk 27]
```diff
@@ -104,12 +119,93 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
...
+    def evaluate(self, x, y, t):
+        """
+        Return the longitudinal envelope.
+
+        Parameters
+        ----------
+        t : ndarrays of floats
+            Define longitudinal points on which to evaluate the envelope
+
+        x,y : ndarrays of floats, necessray if spatio-temperal coulping exists
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
        x,y : ndarrays of floats, necessary if spatio-temporal coupling exists
```

File: lasy/utils/laser_utils.py [hunk 28]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    Returns
    -------
```

File: lasy/utils/laser_utils.py [hunk 29]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
+
+    Return
+    ----------
+    pft_x, pft_y : Pulse front tilt in :math:` p=\frac{dt_0}{dx}` (second * meter^-1).
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    pft_x, pft_y : Pulse front tilt in :math:`p=\frac{dt_0}{dx}` (second * meter^-1).
```

File: lasy/utils/laser_utils.py [hunk 30]
```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
...
+    """
+    assert dim == "xyt", "No pulse front tilt for axis-symmetric dimension."
```
Comment (MaxThevenet) [inline lasy/utils/laser_utils.py]:
```suggestion
    assert dim == "xyt", "No pulse front tilt for cylindrical symmetry."
```

**Review by MaxThevenet — COMMENTED** [review]
Looks fantastic! Just minor comments.

File: lasy/profiles/gaussian_profile.py [hunk 31]
```diff
@@ -61,6 +62,20 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
     z_foc : float (in meter), optional
         Position of the focal plane. (The laser pulse is initialized at `z=0`.)
 
+    phi2 : float (in second^2), optional (default: '0')
+        The group-delay dispersion defined as :math:`\phi^{(2)} = \frac{dt_0}{d\omega}. Here :math:`t_0` is the temporal position of this frequency component.
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
        The group-delay dispersion defined as :math:`\phi^{(2)} = \frac{dt_0}{d\omega}`. Here :math:`t_0` is the temporal position of this frequency component.
```

File: lasy/profiles/gaussian_profile.py [hunk 32]
```diff
@@ -61,6 +62,20 @@ class GaussianProfile(CombinedLongitudinalTransverseProfile):
...
+
+    beta : float (in second), optional (default: '0')
+        The angular dispersion defined as :math:`\beta = \frac{d\theta_0}{d\omega}`. Here :math:`\theta_0` is the propagation angle of this frequency component.
+
+    zeta : float (in meter * second), optional (default: '0')
+        A spatial chirp defined as: math:`\zeta = \frac{dx_0}{d\omega}`. Here :math:`x_0` is the transverse beam center position of this frequency component.
```
Comment (MaxThevenet) [inline lasy/profiles/gaussian_profile.py]:
```suggestion
        A spatial chirp defined as :math:`\zeta = \frac{dx_0}{d\omega}`. Here :math:`x_0` is the transverse beam center position of this frequency component.
```

**Review by MaxThevenet — APPROVED** [review]
\o/

**Comment by spencerjolly** [comment]
Nice work!

**Pull request merged**